autogluon/task/forecasting/dataset.py

```diff
@@ -0,0 +1,168 @@
+from gluonts.dataset.repository.datasets import get_dataset, dataset_recipes
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.common import ListDataset
+import pandas as pd
+
+__all__ = ["ForecastingDataset", "transform_tabular_to_gluonts_dataset", "gluonts_builtin_datasets"]
+
+
+def gluonts_builtin_datasets():
+    return list(dataset_recipes.keys())
+
+
+def rebuild_tabular(X, index_column, date_column, target_column):
+    """
```

Comment:
If `index_column` is None, can we assume there is only 1 time-series, and thus make this argument optional?